frontend/src/viewer/components/path-element.ts

```diff
@@ -333,13 +359,116 @@ export class PathCtrlElement extends connect(store)(LitElement) {
     }
   }
 
+  protected launchScoring(): void {
+    if (this.tracks && this.currentTrack !== null && this.tracks[this.currentTrack]) {
+      const fixes: BRecord[] = [];
+      for (let i = 0; i < this.tracks[this.currentTrack].fixes.lat.length; i++)
+        // Keep this to the bare minimum needed for igc-xc-score
+        fixes[i] = {
+          timestamp: this.tracks[this.currentTrack].fixes.ts[i],
+          latitude: this.tracks[this.currentTrack].fixes.lat[i],
+          longitude: this.tracks[this.currentTrack].fixes.lon[i],
+          pressureAltitude: this.tracks[this.currentTrack].fixes.alt[i],
+          gpsAltitude: this.tracks[this.currentTrack].fixes.alt[i],
+          valid: true,
+          extensions: {} as RecordExtensions,
+          fixAccuracy: null,
+          time: '',
+          enl: null
+        };
+      if (this.worker !== null)
+        this.worker.terminate();
+      this.worker = new Worker('js/xc-score-worker.js');
+      this.worker.onmessage = this.updateScore.bind(this);
+      this.worker.postMessage({ msg: 'xc-score-start', flight: JSON.stringify(fixes), league: this.league });
```

Comment:
It might be nice to use a protobuf (ArrayBuffer) instead of JSON.
It is faster to encode/decode and less memory intensive (does not encoded the field names).
The only thing is that we would need the [en/de]coding logic on the worker side which is ~3kB - probably not a big deal once the worker is cached.